django/views/debug.py

```diff
@@ -259,6 +264,10 @@ def __init__(self, request, exc_type, exc_value, tb, is_email=False):
         self.template_info = getattr(self.exc_value, 'template_debug', None)
         self.template_does_not_exist = False
         self.postmortem = None
+        if not hasattr(self, 'html_template_path'):
+            self.html_template_path = _builtin_template_path('technical_500.html')
+        if not hasattr(self, 'text_template_path'):
+            self.text_template_path = _builtin_template_path('technical_500.txt')
```

Comment:
I'm not convinced about this change is a bit backward incompatible, e.g.
```python
>>> from django.views.debug import ExceptionReporter
>>> ExceptionReporter.html_template_path
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: type object 'ExceptionReporter' has no attribute 'html_template_path'
```
Can we use a `@classproperty` (with `@lru_cache(maxsize=1)`) to keep the current behavior? :thinking:

Reply:
Ah yes, sorry about this one. I knew it wouldn't be perfectly backward compatible when I wrote it. At the time, it was before the 3.2 feature freeze, and `ExceptionReporter`'s template paths were still new (#13531). Now that this patch is targeting 4.0, I can see we need to be more careful.

In 144b0ba8ee8b8d24cea6b766dd12ca6dcc67fcfc I added your [test](https://github.com/django/django/pull/13841/files#diff-57d9651af19f1a34c685f3f561efc7fc25fabb143729640d954f1ed929770737R929-R934) and implemented `ExceptionReporter.{html,text}_template_path` with `@classproperty` and `@lru_cache(maxsize=1)` as you suggested.

However, then I realized that creating a subclass `K` of `ExceptionReporter` and accessing `K.text_template_path` would evict `ExceptionReporter.text_template_path` from the `functools.lru_cache`, thereby defeating the purpose of using the cache in the first place. (As I understand it, the intended use case of `ExceptionReporter.text_template_path` is for easy subclassing of `ExceptionReporter`.) In aa93375963d952c3485511955cf6f536f7fbbc8e, I [amended the test](https://github.com/django/django/pull/13841/files#diff-57d9651af19f1a34c685f3f561efc7fc25fabb143729640d954f1ed929770737R936-R941) and re-implemented the caching manually so that `K.text_template_path is
ExceptionReporter.text_template_path` holds. 

If you prefer the `@lru_cache(maxsize=1)` implementation, it's easy enough to revert the latter commit.